deltacat/compute/compactor_v2/utils/exception_handler.py

```diff
@@ -0,0 +1,138 @@
+    elif isinstance(e, AssertionError):
+        _handle_assertion_error(e)
+    elif isinstance(e, DaftCoreException):
+        _handle_daft_error(e)
+    else:
+        raise e
+
+
+def _get_ray_task_id_and_node_ip():
+    task_id = get_current_ray_task_id()
+    node_ip = ray.util.get_node_ip_address()
+    return task_id, node_ip
+
+
+def _handle_ray_error(e: Exception):
+    task_id, node_ip = _get_ray_task_id_and_node_ip()
+    if isinstance(e, RuntimeEnvSetupError):
+        raise DependencyRayRuntimeSetupError(
+            msg=f"Ray failed to setup runtime env while executing task:{task_id} on node ip:{node_ip}",
+            task_id=task_id,
+            node_ip=node_ip,
+        ) from e
+    elif isinstance(e, WorkerCrashedError) or isinstance(e, NodeDiedError):
+        raise DependencyRayWorkerDiedError(
+            msg=f"Ray worker died unexpectedly while executing task:{task_id} on node ip:{node_ip}.",
+            task_id=task_id,
+            node_ip=node_ip,
+        ) from e
+    elif isinstance(e, OutOfMemoryError):
+        raise DependencyRayOutOfMemoryError(
+            msg=f"Ray worker Out Of Memory while executing task: {task_id} on node ip:{node_ip}.",
+            task_id=task_id,
+            node_ip=node_ip,
+        ) from e
+    else:
+        raise DependencyRayError(
+            msg=f"Dependency Ray error occurred while executing task:{task_id} on node ip:{node_ip}",
+            task_id=task_id,
+            node_ip=node_ip,
+        ) from e
+
+
+def _handle_dependency_pyarrow_error(e: Exception):
+    task_id, node_ip = _get_ray_task_id_and_node_ip()
+    if isinstance(e, ArrowInvalid):
+        raise DependencyPyarrowInvalidError(
+            msg=f"Pyarrow Invalid error occurred while executing task:{task_id} on node ip:{node_ip}.",
+            task_id=task_id,
+            node_ip=node_ip,
+        ) from e
+    elif isinstance(e, ArrowCapacityError):
+        raise DependencyPyarrowCapacityError(
+            msg=f"Pyarrow Invalid error occurred while executing task:{task_id} on node ip:{node_ip}.",
+            task_id=task_id,
+            node_ip=node_ip,
+        ) from e
+    else:
+        raise DependencyPyarrowError(
+            msg=f"Pyarrow error occurred while executing task:{task_id} on node ip:{node_ip}.",
+            task_id=task_id,
+            node_ip=node_ip,
+        ) from e
+
+
+def _handle_daft_error(e: Exception):
+    task_id, node_ip = _get_ray_task_id_and_node_ip()
+    if isinstance(e, DaftTransientError):
+        raise DependencyDaftTransientError()
+    elif isinstance(e, DaftCoreException):
+        raise DependencyDaftError(
+            msg=f"Daft error occurred while executing task:{task_id} on node ip:{node_ip}.",
+            task_id=task_id,
+            node_ip=node_ip,
+        ) from e
+
+
+def _handle_assertion_error(e: Exception):
+    task_id, node_ip = _get_ray_task_id_and_node_ip()
+    raise GeneralAssertionError(
```

Comment:
This can be classified as `ValidationError` as it would come from assertions in the code.